tests/test_messaging.py

```diff
@@ -505,7 +577,142 @@ def test_android_notification(self):
+                    'default_vibrate_timings': 0,
+                    'default_sound': 1,
+                    'light_settings': {
+                        'color': {
+                            'red': 0.6666666666666666,
+                            'green': 0.7333333333333333,
+                            'blue': 0.8,
+                            'alpha': 0.8666666666666667,
+                        },
+                        'light_on_duration': '0.200000000s',
+                        'light_off_duration': '0.300000000s',
+                    },
+                    'default_light_settings': 0,
+                    'visibility': 'PUBLIC',
+                    'notification_count': 1,
+                },
+            },
+        }
+        check_encoding(msg, expected)
+
+
+class TestLightSettingsEncoder(object):
+
+    def _check_light_settings(self, light_settings):
+        with pytest.raises(ValueError) as excinfo:
+            check_encoding(messaging.Message(
+                topic='topic', android=messaging.AndroidConfig(
+                    notification=messaging.AndroidNotification(
+                        light_settings=light_settings
+                    ))))
+        return excinfo
+
+    @pytest.mark.parametrize('data', NON_OBJECT_ARGS)
+    def test_invalid_light_settings(self, data):
+        with pytest.raises(ValueError) as excinfo:
+            check_encoding(messaging.Message(
+                topic='topic', android=messaging.AndroidConfig(
+                    notification=messaging.AndroidNotification(
+                        light_settings=data
+                    ))))
+        expected = 'AndroidNotification.light_settings must be an instance of LightSettings class.'
+        assert str(excinfo.value) == expected
+
+    def test_no_color(self):
+        light_settings = messaging.LightSettings(light_on_duration_millis=200,
+                                                 light_off_duration_millis=200)
+        excinfo = self._check_light_settings(light_settings)
+        expected = 'LightSettings.color is required.'
+        assert str(excinfo.value) == expected
+
+    def test_no_light_on_duration_millis(self):
+        light_settings = messaging.LightSettings(color='#aabbcc',
+                                                 light_off_duration_millis=200)
+        excinfo = self._check_light_settings(light_settings)
+        expected = 'LightSettings.light_on_duration_millis is required.'
+        assert str(excinfo.value) == expected
+
+    def test_no_light_off_duration_millis(self):
+        light_settings = messaging.LightSettings(color='#aabbcc',
+                                                 light_on_duration_millis=200)
+        excinfo = self._check_light_settings(light_settings)
+        expected = 'LightSettings.light_off_duration_millis is required.'
+        assert str(excinfo.value) == expected
+
+    @pytest.mark.parametrize('data', ['1.23s', list(), tuple(), dict(), -1.23])
```

Comment:
Perhaps define a class-level constant for the list.

Reply:
Named it `NON_UNUM_ARGS` as in `non-unsigned-numbers`. Open to better naming suggestions :)